test/account-abstraction/LightAccountVerificationV1.test.ts

```diff
@@ -0,0 +1,135 @@
+    initializeCalldata,
+    salt,
+  );
+
+  return ConcreteLightAccountVerification__factory.connect(
+    predictedLightAccountVerificationAddress,
+    deployer,
+  );
+}
+
+describe('LightAccountVerificationV1', function () {
+  // contracts
+  let concreteVerification: ConcreteLightAccountVerification;
+  let mockLightAccount: MockLightAccount;
+  let mockInvalidLightAccount: MockInvalidLightAccount;
+  let mockLightAccountFactory: MockLightAccountFactory;
+
+  // signers
+  let deployer: SignerWithAddress;
+  let owner: SignerWithAddress;
+
+  beforeEach(async function () {
+    // Get signers
+    [deployer, owner] = await ethers.getSigners();
+
+    // Deploy MockLightAccount
+    mockLightAccount = await new MockLightAccount__factory(deployer).deploy(owner.address);
+
+    // Deploy MockInvalidLightAccount
+    mockInvalidLightAccount = await new MockInvalidLightAccount__factory(deployer).deploy();
+
+    // Deploy MockLightAccountFactory
+    mockLightAccountFactory = await new MockLightAccountFactory__factory(deployer).deploy();
+
+    // Deploy MockLightAccountVerification
+    const concreteVerificationImplementation = await new ConcreteLightAccountVerification__factory(
+      deployer,
+    ).deploy();
+
+    concreteVerification = await deployLightAccountVerification(
+      deployer,
+      concreteVerificationImplementation,
+      mockLightAccountFactory.target.toString(),
+    );
+  });
+
+  describe('verifyLightAccount', function () {
+    describe('valid light accounts', function () {
+      it('should verify valid light accounts', async function () {
+        // set up the mock factory contract to return the correct address to `verifyLightAccount`
+        await mockLightAccountFactory.setAccountAddress(
+          await mockLightAccount.owner(),
+          0n,
+          await mockLightAccount.getAddress(),
+        );
+
+        void expect(
+          await concreteVerification.verifyLightAccountPublic(await mockLightAccount.getAddress()),
+        ).to.be.true;
+      });
+    });
+
+    describe('invalid light accounts', function () {
+      describe('non-contracts', function () {
+        it('should return false for non-contract addresses', async function () {
+          const randomAddress = ethers.Wallet.createRandom().address;
+
+          // Should return false since the address won't have the owner() function
+          void expect(await concreteVerification.verifyLightAccountPublic(randomAddress)).to.be
+            .false;
+        });
+      });
+
+      describe('contracts', function () {
+        it('should return false for invalid light accounts that do implement the ILightAccount interface (owner())', async function () {
+          // not calling "setAccountAddress", so the LightAccountFactory will always
+          // return the zero address when calling `getAddress` for a given owner and salt
+          // (which is implemented in the LightAccountVerification verifyLightAccount function).
```

Comment:
Then this comment would be more intuitive, because the test's (mock) light account factory hasn't "deployed" the `mockLightAccount`